internal/fsutil/path.go

```diff
@@ -0,0 +1,47 @@
+package fsutil
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// These functions exists because we work with slash terminated directory paths
+// that come from deb package tarballs but standard library path functions
+// treat slash terminated paths as unclean.
+
+// Like filepath.Dir() but trailing slash on input is ignored and the return
+// value always includes trailing slash.
+//
+// Comparison of filepath.Dir() with fsutil.Dir():
+//
+//	filepath.Dir("/foo/bar/")	== "/foo/bar"
+//	filepath.Dir("/foo/bar")	== "/foo"
+//
+//	fsutil.Dir("/foo/bar")		== "/foo/"
+//	fsutil.Dir("/foo/bar/")		== "/foo/"
+func Dir(path string) string {
+	parent := filepath.Dir(filepath.Clean(path))
+	if parent != "/" {
+		parent += "/"
+	}
+	return parent
+}
+
+// Like filepath.Clean() but keeps trailing slash.
+//
+// Comparison of filepath.Clean() with fsutil.Clean():
+//
+//	filepath.Clean("/foo/bar")	== "/foo/bar"
+//	filepath.Clean("/foo/bar/")	== "/foo/bar"
+//	filepath.Clean("/foo/bar/.//)	== "/foo/bar"
+//
+//	fsutil.Clean("/foo/bar")	== "/foo/bar"
```

Comment:
That would turn regular file path `/foo/bar` into directory path `/foo/bar/`. Please see the usage of `fsutil.Clean()` in this commit. If I were to only use it for directory paths, it'd defeat the purpose of abstracting away the convention of ending every directory path with slash. I.e. instead of
```
path := fsutil.Clean()
```
now I'd have
```
var path string
if strings.HasSuffix("/") {
  path = fsutil.Clean(path)
} else {
  path = filepath.Clean(path)
}
```
which doesn't seem like an improvement over existing code. See the example in the commit message where we do this wrong in one place and Gustavo's comment I linked above about the path format assumption.

Reply:
I missed the fact that both files and directories needs to be handled. Thanks for explaining this to me.